zio-kafka/src/main/scala/zio/kafka/admin/AdminClient.scala

```diff
@@ -628,33 +628,45 @@ object AdminClient {
      */
     override def listConsumerGroupOffsets(
       groupSpecs: Map[String, ListConsumerGroupOffsetsSpec]
-    ): Task[Map[TopicPartition, OffsetAndMetadata]] =
+    ): Task[Map[String, Map[TopicPartition, OffsetAndMetadata]]] =
       fromKafkaFuture {
         ZIO.attemptBlocking(
           adminClient
             .listConsumerGroupOffsets(groupSpecs.map { case (groupId, offsetsSpec) =>
               (groupId, offsetsSpec.asJava)
             }.asJava)
-            .partitionsToOffsetAndMetadata()
+            .all()
         )
       }
-        .map(_.asScala.filter { case (_, om) => om ne null }.toMap.bimap(TopicPartition(_), OffsetAndMetadata(_)))
+        .map(
+          _.asScala.view
+            .mapValues(_.asScala.filter { case (_, om) =>
+              om ne null
+            }.toMap.bimap(TopicPartition(_), OffsetAndMetadata(_)))
+            .toMap
+        )
 
     override def listConsumerGroupOffsets(
       groupSpecs: Map[String, ListConsumerGroupOffsetsSpec],
       options: ListConsumerGroupOffsetsOptions
-    ): Task[Map[TopicPartition, OffsetAndMetadata]] =
+    ): Task[Map[String, Map[TopicPartition, OffsetAndMetadata]]] =
```

Comment:
what about defining a type alias to make it clear String is a TopicName?

Reply:
The fact that you think this String is a topic name instead of a group ID strengthens your point :smile: 
To keep the code consistent, it would require to change many methods. I suggest we make this change in another PR and try to port it upstream.